client-index/src/auto_sync_data.rs

```diff
@@ -0,0 +1,114 @@
+//! auto sync sharing state
+//! shared between network and core threads
+
+use chain_core::state::account::StakedStateAddress;
+use client_common::{PrivateKey, PublicKey};
+use serde::{Deserialize, Serialize};
+use std::sync::{Arc, Mutex};
+use websocket::OwnedMessage;
+
+/// give add wallet command via this queue
+pub type AutoSyncQueue = std::sync::mpsc::Sender<OwnedMessage>;
+
+#[derive(Clone, Debug)]
+/// auto sync internal data
+pub struct AutoSyncData {
+    /// normalized: 0.0 ~ 1.0
+    pub progress: f64,
+    /// current
+    pub current_height: u64,
+    /// max height
+    pub max_height: u64,
+    /// current syncing wallet name
+    pub wallet: String,
+    /// send queue
+    pub send_queue: Option<std::sync::mpsc::Sender<OwnedMessage>>,
+}
+
+impl Default for AutoSyncData {
```

Comment:
You can add `#[derive(Default)]` instead of implementing this.

Reply:
wow. so convenient